op-program/client/driver/driver.go

```diff
@@ -43,9 +43,8 @@ func NewDriver(logger log.Logger, cfg *rollup.Config, depSet derive.DependencySe
 	pipelineDeriver := derive.NewPipelineDeriver(context.Background(), pipeline)
 	pipelineDeriver.AttachEmitter(d)
 
-	ec := engine.NewEngineController(l2Source, logger, metrics.NoopMetrics, cfg, &sync.Config{SyncMode: sync.CLSync}, d)
-	engineDeriv := engine.NewEngDeriver(logger, context.Background(), cfg, metrics.NoopMetrics, ec)
-	engineDeriv.AttachEmitter(d)
+	ec := engine.NewEngineControllerWithEventFields(context.Background(), l2Source, logger, metrics.NoopMetrics, cfg, &sync.Config{SyncMode: sync.CLSync}, d)
+	ec.AttachEmitter(d)
```

Comment:
I believe this `AttachEmitter` call is unnecessary. I don't think `EngineController` needs to implement `AttachEmitter` at all in fact.

Edit: unless the emitter is changed somewhere, but I don't think that's the case?